// @argent/x-ui/dist/src/components/logos-deprecated/index.js
import { default as r } from "./ArgentXLogo.js";
import { default as a } from "./ArgentXLogoFull.js";
import { default as s } from "./AspectLogo.js";
import { default as l } from "./AvnuLogo.js";
import { default as m } from "./BanxaLogo.js";
import { default as u } from "./BlastLogo.js";
import { default as L } from "./BriqLogo.js";
import { default as i } from "./CoinbaseLogo.js";
import { default as y } from "./DiscordLogo.js";
import { default as F } from "./ElementLogo.js";
import { default as b } from "./EmptyNftLogo.js";
import { default as h } from "./EthereumLogo.js";
import { default as B } from "./FlexLogo.js";
import { default as I } from "./FrenslandsLogo.js";
import { default as q } from "./GithubLogo.js";
import { default as T } from "./ImageLogo.js";
import { default as C } from "./InfluenceLogo.js";
import { default as G } from "./JediswapLogo.js";
import { default as M } from "./LayerswapLogo.js";
import { default as O } from "./LedgerLogo.js";
import { default as U } from "./MintsquareLogo.js";
import { default as Z } from "./OrbiterLogo.js";
import { default as z } from "./PyramidLogo.js";
import { default as K } from "./RampLogo.js";
import { default as W } from "./RhinoFiLogo.js";
import { default as _ } from "./SpokLogo.js";
import { default as oo } from "./StarkgateLogo.js";
import { default as ro } from "./StarknetIdLogo.js";
import { default as ao } from "./StarknetLogo.js";
import { default as so } from "./TopperLogo.js";
import { default as po } from "./TwitterLogo.js";
import { default as go } from "./UnframedLogo.js";
import { default as xo } from "./VoyagerLogo.js";
import { default as no } from "./XLogo.js";
import { default as ko } from "./ZklendLogo.js";
export {
  r as ArgentXLogo,
  a as ArgentXLogoFull,
  s as AspectLogo,
  l as AvnuLogo,
  m as BanxaLogo,
  u as BlastLogo,
  L as BriqLogo,
  i as CoinbaseLogo,
  y as DiscordLogo,
  F as ElementLogo,
  b as EmptyNftLogo,
  h as EthereumLogo,
  B as FlexLogo,
  I as FrenslandsLogo,
  q as GithubLogo,
  T as ImageLogo,
  C as InfluenceLogo,
  G as JediswapLogo,
  M as LayerswapLogo,
  O as LedgerLogo,
  U as MintsquareLogo,
  Z as OrbiterLogo,
  z as PyramidLogo,
  K as RampLogo,
  W as RhinoFiLogo,
  _ as SpokLogo,
  oo as StarkgateLogo,
  ro as StarknetIdLogo,
  ao as StarknetLogo,
  so as TopperLogo,
  po as TwitterLogo,
  go as UnframedLogo,
  xo as VoyagerLogo,
  no as XLogo,
  ko as ZklendLogo
};
