// @argent/x-ui/dist/src/components/transactions/NftDetails.js
import { jsxs as m, jsx as e } from "react/jsx-runtime";
import { generateAvatarImage as x, getColor as C, getNftPicture as T, prettifyTokenAmount as w, prettifyCurrencyValue as F, bigDecimal as W } from "@argent/x-shared";
import { useColorMode as I, AccordionButton as P, Flex as b, Image as U } from "@chakra-ui/react";
import { useMemo as _, useCallback as j } from "react";
import v from "../icons/WarningCircleSecondaryIcon.js";
import { TextWithAmount as A } from "../TextWithAmount.js";
import { P3 as h, P2 as B, Label2Bold as M } from "../typography.generated.js";
const c = "Unknown NFT", q = ({ contractAddress: o, safe: f, dataIndex: p, totalData: s, isDisabled: u, isDefaultNetwork: N, amount: a, usdValue: l, nft: r, collection: n }) => {
  const { colorMode: d } = I(), g = _(() => d === "dark", [d]), i = j((t, k) => x(k ?? c, {
    background: C(t)
  }), []), y = _(() => {
    if (r && r.name)
      return {
        name: r.name,
        description: r.description,
        image_url_copy: r.image_url_copy ?? r.image_uri ?? i(o, r.name)
      };
    if (n) {
      const t = n.name ?? "Unknown Collection";
      return {
        name: `${t} NFT`,
        description: `${t} NFT`,
        image_url_copy: n.imageUri ?? i(o)
      };
    }
    return {
      name: c,
      description: c,
      image_url_copy: i(o)
    };
  }, [o, n, r, i]);
  return /* @__PURE__ */ m(P, { display: "flex", width: "100%", justifyContent: "space-between", outline: "none", px: "3", pb: p !== s - 1 ? "3" : "3.5", _expanded: {
    backgroundColor: g ? "neutrals.700" : "gray.50",
    pb: "3.5"
  }, disabled: u, _disabled: {
    cursor: "auto",
    opacity: 1
  }, _hover: {
    backgroundColor: u ? "" : g ? "neutrals.700" : "gray.50",
    borderBottomRadius: p === s - 1 ? "xl" : "0"
  }, children: [
    /* @__PURE__ */ m(b, { alignItems: "center", gap: "2", children: [
      /* @__PURE__ */ e(U, { src: T({ image_url_copy: y.image_url_copy }) || "", w: "5", h: "5", borderRadius: "base" }),
      /* @__PURE__ */ e(h, { fontWeight: "bold", color: "text-primary", children: y.name }),
      !f && /* @__PURE__ */ e(B, { color: "error.500", fontWeight: "bold", mt: "0.25", children: /* @__PURE__ */ e(v, {}) })
    ] }),
    /* @__PURE__ */ m(b, { direction: "column", gap: "0.5", alignItems: "flex-end", children: [
      /* @__PURE__ */ e(A, { amount: a, children: /* @__PURE__ */ e(h, { color: a < 0n ? "error.500" : "secondary.500", fontWeight: "bold", children: w({
        amount: a,
        decimals: 0,
        symbol: "NFT",
        showPlusSign: !0
      }) }) }),
      N && !!l && l !== "0" && /* @__PURE__ */ e(M, { color: "neutrals.300", children: F(W.parseCurrencyAbs(l).toString()) })
    ] })
  ] });
};
export {
  q as NftDetails
};
