// @argent/x-ui/dist/theme.js
import { theme as r } from "./src/features/theme/theme.js";
import { theme as m } from "./src/features/theme/theme.web.js";
import { theme as p } from "./src/features/theme/theme.mobile.js";
import { scrollbarStyle as l, scrollbarStyleThin as x } from "./src/features/theme/scrollbarStyle.js";
import { SetDarkMode as n } from "./src/features/theme/SetDarkMode.js";
import { ThemeProvider as s } from "./src/features/theme/ThemeProvider.js";
import { ThemeProviderWeb as b } from "./src/features/theme/ThemeProviderWeb.js";
import { buttonBaseStyle as d, buttonHoverStyle as y } from "./src/features/theme/buttonTheme.js";
import { typographyStyles as w } from "./src/features/theme/typographyStyles.generated.js";
import { colors as u } from "./src/features/theme/tailwind/colors.generated.js";
import { fontSize as F } from "./src/features/theme/tailwind/fontSize.mobile.generated.js";
import { fontSize as P } from "./src/features/theme/tailwind/fontSize.plugin.generated.js";
import { fontSize as g } from "./src/features/theme/tailwind/fontSize.web.generated.js";
import { pxToRem as B } from "./src/features/theme/utilities/pxToRem.js";
import { pxToEm as D } from "./src/features/theme/utilities/pxToEm.js";
export {
  n as SetDarkMode,
  s as ThemeProvider,
  b as ThemeProviderWeb,
  d as buttonBaseStyle,
  y as buttonHoverStyle,
  D as pxToEm,
  B as pxToRem,
  l as scrollbarStyle,
  x as scrollbarStyleThin,
  u as tailwindThemeColors,
  F as tailwindThemeFontSizeMobile,
  P as tailwindThemeFontSizePlugin,
  g as tailwindThemeFontSizeWeb,
  r as theme,
  p as themeMobile,
  m as themeWeb,
  w as typographyStyles
};
