// @argent/x-ui/dist/src/components/CompoundInput/components.js
import { jsxs as l, jsx as o, Fragment as g } from "react/jsx-runtime";
import { prettifyCurrencyValue as b } from "@argent/x-shared";
import { Flex as a, Input as x, Button as u, Circle as C } from "@chakra-ui/react";
import { useCompoundInputContext as d } from "./context.js";
import { getRadius as y } from "./utils.js";
import f from "../icons/ChevronDownPrimaryIcon.js";
import I from "../icons/InfoCircleSecondaryIcon.js";
import { LoadingPulse as v } from "../LoadingPulse.js";
import { TokenIcon as m } from "../TokenIcon.js";
import { P3 as h, H5 as p, L2Bold as s, L1Bold as w } from "../typography.generated.js";
function S({ children: t, ...n }) {
  return /* @__PURE__ */ o(a, { children: /* @__PURE__ */ o(h, { fontWeight: "bold", fontSize: "sm", ...n, children: t }) });
}
function W({ ...t }) {
  const { id: n, amount: e, isDisabled: r, isLoading: i, error: c } = d();
  return e ? /* @__PURE__ */ o(v, { isLoading: i && !e.value, mr: "3", children: /* @__PURE__ */ o(x, { placeholder: "0", color: c ? "text-danger" : "text-primary", p: "0", minH: "min-content", backgroundColor: "transparent", variant: "flat", fontSize: "2xl", fontWeight: "600", lineHeight: "6.5", value: e.value, onChange: e.onChange, isDisabled: r, isReadOnly: e.isReadOnly, _disabled: { opacity: 1, cursor: "default" }, _hover: {
    backgroundColor: "transparent"
  }, _focus: {
    boxShadow: "none",
    backgroundColor: "transparent"
  }, _readOnly: {
    pointerEvents: "none",
    border: "none",
    cursor: "default"
  }, tabIndex: e.isReadOnly ? -1 : 1, "data-testid": n, ...t }) }) : null;
}
function z({ children: t, showChevron: n = !0, ...e }) {
  var i, c;
  const { token: r } = d();
  return /* @__PURE__ */ o(a, { alignItems: "center", children: /* @__PURE__ */ o(u, { px: n ? r != null && r.selected ? "1" : "3" : "2", py: n ? "1" : "1.5", height: "min-content", minH: "min-content", width: "min-content", rounded: "full", onClick: r == null ? void 0 : r.onClick, leftIcon: r != null && r.selected ? /* @__PURE__ */ o(m, { name: ((i = r == null ? void 0 : r.selected) == null ? void 0 : i.name) || "?", url: (c = r == null ? void 0 : r.selected) == null ? void 0 : c.iconUrl, size: "5", borderRadius: "50%" }) : void 0, rightIcon: n ? /* @__PURE__ */ o(f, {}) : void 0, fontWeight: "500", backgroundColor: "surface-sunken", ...e, children: /* @__PURE__ */ o(p, { children: r != null && r.selected ? r.selected.symbol || r.selected.name : t }) }) });
}
function B({ children: t, ...n }) {
  var r, i;
  const { token: e } = d();
  return /* @__PURE__ */ o(a, { alignItems: "center", width: "100%", children: /* @__PURE__ */ l(u, { p: "0", gap: "2", width: "100%", rounded: "full", onClick: e == null ? void 0 : e.onClick, fontWeight: "500", backgroundColor: "transparent", _hover: {
    backgroundColor: "transparent"
  }, _active: {
    transform: "scale(1)"
  }, ...n, children: [
    e === void 0 || e != null && e.selected || !(e != null && e.PlaceholderIcon) ? /* @__PURE__ */ o(m, { name: ((r = e == null ? void 0 : e.selected) == null ? void 0 : r.name) || "?", url: (i = e == null ? void 0 : e.selected) == null ? void 0 : i.iconUrl, size: "8", borderRadius: "50%" }) : /* @__PURE__ */ o(C, { backgroundColor: "icon-background", size: "8", children: /* @__PURE__ */ o(e.PlaceholderIcon, { color: "icon-secondary" }) }),
    /* @__PURE__ */ o(p, { color: e != null && e.selected ? "" : "text-subtle", children: e != null && e.selected ? e.selected.name : t }),
    /* @__PURE__ */ o(f, { ml: "auto" })
  ] }) });
}
function R({ children: t, ...n }) {
  const { amount: e } = d();
  if (e != null && e.max)
    return /* @__PURE__ */ o(s, { onClick: e.max.onClick, cursor: "pointer", role: "button", color: "text-brand", fontWeight: "600", fontSize: "xs", whiteSpace: "nowrap", ...n, children: t });
}
function _({ ...t }) {
  var r, i, c;
  const { amount: n } = d(), e = ((r = n == null ? void 0 : n.fiat) == null ? void 0 : r.value) || "0";
  return /* @__PURE__ */ l(s, { color: "text-secondary", fontSize: "sm", fontWeight: "400", ...t, children: [
    ((i = n == null ? void 0 : n.fiat) == null ? void 0 : i.approx) && /* @__PURE__ */ o(g, { children: "≈ " }),
    b(e, (c = n == null ? void 0 : n.fiat) == null ? void 0 : c.symbol)
  ] });
}
function D({ ...t }) {
  var e;
  const { token: n } = d();
  if (((e = n == null ? void 0 : n.selected) == null ? void 0 : e.balance) !== void 0)
    return /* @__PURE__ */ l(s, { color: "text-secondary", fontSize: "sm", fontWeight: "400", textAlign: "right", ...t, children: [
      "Balance: ",
      n == null ? void 0 : n.selected.balance
    ] });
}
function L({ ...t }) {
  const { error: n } = d();
  if (n)
    return /* @__PURE__ */ l(w, { as: a, alignItems: "center", gap: "0.5", color: "text-danger", ...t, children: [
      /* @__PURE__ */ o(I, { w: "4", h: "4", color: "inherit" }),
      /* @__PURE__ */ o(h, { fontWeight: "bold", children: n })
    ] });
}
function P({ children: t, ...n }) {
  const { position: e } = d();
  return /* @__PURE__ */ o(a, { position: "relative", flexDirection: "column", gap: "2", padding: "4", backgroundColor: "surface-elevated", width: "100%", ...y(e, "lg"), ...n, children: t });
}
function q({ title: t, selectLabel: n = "Select token", ...e }) {
  var i, c;
  const { amount: r } = d();
  return /* @__PURE__ */ l(P, { ...e, children: [
    /* @__PURE__ */ o(S, { children: t }),
    /* @__PURE__ */ l(a, { flexDirection: "column", gap: "1", children: [
      /* @__PURE__ */ l(a, { justifyContent: "space-between", alignItems: "center", children: [
        r && /* @__PURE__ */ o(W, {}),
        r ? /* @__PURE__ */ o(z, { showChevron: !1, children: n }) : /* @__PURE__ */ o(B, { children: n })
      ] }),
      r && /* @__PURE__ */ l(a, { justifyContent: "space-between", children: [
        /* @__PURE__ */ l(a, { gap: "1", children: [
          ((i = r.fiat) == null ? void 0 : i.value) && /* @__PURE__ */ o(_, {}),
          ((c = r.max) == null ? void 0 : c.show) && /* @__PURE__ */ o(R, { children: "Use max" })
        ] }),
        /* @__PURE__ */ o(D, { marginLeft: "auto" })
      ] })
    ] }),
    /* @__PURE__ */ o(L, {})
  ] });
}
export {
  P as Container,
  q as Default,
  L as Error,
  _ as FiatValue,
  W as InputAmount,
  z as InputToken,
  R as MaxButton,
  S as Title,
  D as TokenBalance
};
