// @argent/x-ui/dist/src/components/NavigationBar.js
import { jsx as o, jsxs as d, Fragment as b } from "react/jsx-runtime";
import { Button as g, chakra as w, Flex as h, Center as e, SkeletonCircle as p, SkeletonText as x } from "@chakra-ui/react";
import * as B from "./icons/index.js";
import "./Typography.js";
import { useNavigateBack as u } from "../hooks/useNavigateBack.js";
import { H5 as k } from "./typography.generated.js";
const { PlusSecondaryIcon: S, CrossSecondaryIcon: y, ArrowLeftPrimaryIcon: C } = B, i = 14, v = w(h, {
  baseStyle: {
    alignItems: "center",
    bottom: "initial",
    h: i,
    px: 3,
    py: 2,
    transitionProperty: "background",
    transitionDuration: "fast",
    flexShrink: 0,
    zIndex: 1
  }
}), c = ({ children: t, ...n }) => /* @__PURE__ */ o(g, { color: "icon-secondary", colorScheme: "transparent", padding: "1.5", fontSize: "xl", size: "auto", rounded: "full", ...n, children: t }), P = (t) => {
  const n = u();
  return /* @__PURE__ */ o(c, { "aria-label": "Back", onClick: n, ...t, children: /* @__PURE__ */ o(C, {}) });
}, F = (t) => {
  const n = u();
  return /* @__PURE__ */ o(c, { "aria-label": "Close", onClick: n, ...t, children: /* @__PURE__ */ o(y, {}) });
}, H = (t) => /* @__PURE__ */ o(c, { "aria-label": "add", ...t, children: /* @__PURE__ */ o(S, {}) }), L = ({ isAbsolute: t, leftButton: n, rightButton: l, title: r, scroll: a, children: m, ...f }) => {
  const s = ((a == null ? void 0 : a.scrollTop) ?? 0) <= 16;
  return /* @__PURE__ */ d(v, { bg: s ? "transparent" : "surface-default", boxShadow: s ? "none" : "menu", position: t ? "absolute" : "relative", w: "full", ...f, children: [
    r && /* @__PURE__ */ o(e, { position: "absolute", left: i, right: i, top: 0, bottom: 0, children: typeof r == "string" ? /* @__PURE__ */ o(k, { w: "full", overflow: "hidden", textOverflow: "ellipsis", whiteSpace: "nowrap", textAlign: "center", children: r }) : /* @__PURE__ */ o(b, { children: r }) }),
    n && /* @__PURE__ */ o(e, { position: "absolute", left: 3, top: 2, bottom: 2, children: n }),
    l && /* @__PURE__ */ o(e, { position: "absolute", right: 3, top: 2, bottom: 2, children: l }),
    m
  ] });
}, O = () => /* @__PURE__ */ d(h, { alignItems: "center", justifyContent: "space-between", px: 3, py: 2, height: i, children: [
  /* @__PURE__ */ o(e, { w: 8, h: 8, children: /* @__PURE__ */ o(p, { w: 5, h: 5 }) }),
  /* @__PURE__ */ o(x, { h: 2, w: "33.3%", noOfLines: 1 }),
  /* @__PURE__ */ o(e, { w: 8, h: 8, children: /* @__PURE__ */ o(p, { w: 5, h: 5 }) })
] });
export {
  H as BarAddButton,
  P as BarBackButton,
  F as BarCloseButton,
  c as BarIconButton,
  L as NavigationBar,
  i as NavigationBarHeight,
  O as NavigationBarSkeleton
};
