// @argent/x-ui/dist/src/components/typography.generated.js
import { typographyStyles as e } from "../features/theme/typographyStyles.generated.js";
import { chakra as a } from "@chakra-ui/react";
const s = a("span", {
  baseStyle: {
    ...e.B1
  }
}), o = a("span", {
  baseStyle: {
    ...e.B2
  }
}), n = a("span", {
  baseStyle: {
    ...e.B3
  }
}), b = a("span", {
  baseStyle: {
    ...e.Button1
  }
}), y = a("span", {
  baseStyle: {
    ...e.Button2
  }
}), c = a("span", {
  baseStyle: {
    ...e.Button3
  }
}), S = a("h1", {
  baseStyle: {
    ...e.H1
  }
}), d = a("h2", {
  baseStyle: {
    ...e.H2
  }
}), p = a("h3", {
  baseStyle: {
    ...e.H3
  }
}), B = a("h4", {
  baseStyle: {
    ...e.H4
  }
}), r = a("h5", {
  baseStyle: {
    ...e.H5
  }
}), h = a("h3", {
  baseStyle: {
    ...e.H3Medium
  }
}), g = a("h4", {
  baseStyle: {
    ...e.H4Medium
  }
}), i = a("h1", {
  baseStyle: {
    ...e.Heading1
  }
}), H = a("h2", {
  baseStyle: {
    ...e.Heading2
  }
}), P = a("h3", {
  baseStyle: {
    ...e.Heading3
  }
}), L = a("h4", {
  baseStyle: {
    ...e.Heading4
  }
}), u = a("h5", {
  baseStyle: {
    ...e.Heading5
  }
}), m = a("h3", {
  baseStyle: {
    ...e.Heading3Medium
  }
}), M = a("h4", {
  baseStyle: {
    ...e.Heading4Medium
  }
}), f = a("label", {
  baseStyle: {
    ...e.L1
  }
}), v = a("label", {
  baseStyle: {
    ...e.L2
  }
}), T = a("label", {
  baseStyle: {
    ...e.L3
  }
}), k = a("label", {
  baseStyle: {
    ...e.L1Bold
  }
}), x = a("label", {
  baseStyle: {
    ...e.L2Bold
  }
}), j = a("label", {
  baseStyle: {
    ...e.L3Bold
  }
}), q = a("label", {
  baseStyle: {
    ...e.Label1
  }
}), w = a("label", {
  baseStyle: {
    ...e.Label2
  }
}), z = a("label", {
  baseStyle: {
    ...e.Label3
  }
}), A = a("label", {
  baseStyle: {
    ...e.Label1Bold
  }
}), C = a("label", {
  baseStyle: {
    ...e.Label2Bold
  }
}), D = a("label", {
  baseStyle: {
    ...e.Label3Bold
  }
}), E = a("p", {
  baseStyle: {
    ...e.MassiveTitle
  }
}), F = a("p", {
  baseStyle: {
    ...e.P1
  }
}), G = a("p", {
  baseStyle: {
    ...e.P2
  }
}), I = a("p", {
  baseStyle: {
    ...e.P3
  }
}), J = a("p", {
  baseStyle: {
    ...e.P4
  }
}), K = a("p", {
  baseStyle: {
    ...e.P2Bold
  }
}), N = a("p", {
  baseStyle: {
    ...e.P3Bold
  }
}), O = a("p", {
  baseStyle: {
    ...e.P4Bold
  }
}), Q = a("p", {
  baseStyle: {
    ...e.Paragraph1
  }
}), R = a("p", {
  baseStyle: {
    ...e.Paragraph2
  }
}), U = a("p", {
  baseStyle: {
    ...e.Paragraph3
  }
}), V = a("p", {
  baseStyle: {
    ...e.Paragraph4
  }
}), W = a("p", {
  baseStyle: {
    ...e.Paragraph2Bold
  }
}), X = a("p", {
  baseStyle: {
    ...e.Paragraph3Bold
  }
}), Y = a("p", {
  baseStyle: {
    ...e.Paragraph4Bold
  }
});
export {
  s as B1,
  o as B2,
  n as B3,
  b as Button1,
  y as Button2,
  c as Button3,
  S as H1,
  d as H2,
  p as H3,
  h as H3Medium,
  B as H4,
  g as H4Medium,
  r as H5,
  i as Heading1,
  H as Heading2,
  P as Heading3,
  m as Heading3Medium,
  L as Heading4,
  M as Heading4Medium,
  u as Heading5,
  f as L1,
  k as L1Bold,
  v as L2,
  x as L2Bold,
  T as L3,
  j as L3Bold,
  q as Label1,
  A as Label1Bold,
  w as Label2,
  C as Label2Bold,
  z as Label3,
  D as Label3Bold,
  E as MassiveTitle,
  F as P1,
  G as P2,
  K as P2Bold,
  I as P3,
  N as P3Bold,
  J as P4,
  O as P4Bold,
  Q as Paragraph1,
  R as Paragraph2,
  W as Paragraph2Bold,
  U as Paragraph3,
  X as Paragraph3Bold,
  V as Paragraph4,
  Y as Paragraph4Bold
};
