// @argent/x-ui/dist/src/components/CompoundDialogBottomSheet/components/RowList.js
import { jsxs as i, jsx as e, Fragment as S } from "react/jsx-runtime";
import { Flex as m, Tabs as T, Spinner as v, TabPanels as w, TabPanel as F } from "@chakra-ui/react";
import { isObject as I } from "lodash-es";
import { useRef as R, useState as D } from "react";
import { Virtuoso as L } from "react-virtuoso";
import "../../Typography.js";
import { WebSearchInput as P } from "../../WebSearchInput/index.js";
import { H5 as p, P4 as k } from "../../typography.generated.js";
function C(t) {
  return t && I(t[0]) && "title" in t[0] && "rows" in t[0];
}
function z({ rows: t, RowComponent: b, NoResultsComponent: u, loading: s, allowSearch: y = !0, searchRows: f }) {
  const h = C(t), l = {}, o = R([
    ...h ? t.flatMap(({ title: n, rows: a }, r) => {
      let g = 0;
      return r !== 0 && (g = Object.keys(l).reduce((x, j) => Number(j) + Number(x), t[r - 1].rows.length)), l[g] = n, a;
    }) : t
  ]), [c, d] = D(o.current);
  return /* @__PURE__ */ i(m, { flexDirection: "column", gap: "6", children: [
    y && f && /* @__PURE__ */ e(P, { onChange: (n) => {
      d(f(n.target.value, o.current));
    }, onReset: () => d(o.current) }),
    /* @__PURE__ */ i(
      T,
      {
        variant: "unstyled",
        position: "relative",
        isFitted: !0,
        h: { base: "full", sm: "92" },
        overflowY: "auto",
        children: [
          !s && c.length === 0 && /* @__PURE__ */ i(m, { h: "full", w: "full", flexDirection: "column", justifyContent: "center", alignItems: "center", gap: "3", py: "16", children: [
            u && /* @__PURE__ */ e(u, {}),
            !u && /* @__PURE__ */ e(p, { children: "Nothing to select" })
          ] }),
          s && /* @__PURE__ */ i(m, { h: "full", w: "full", flexDirection: "column", justifyContent: "center", alignItems: "center", gap: "3", py: "16", children: [
            /* @__PURE__ */ e(v, {}),
            /* @__PURE__ */ e(p, { children: "Loading..." })
          ] }),
          !s && c.length > 0 && /* @__PURE__ */ e(w, { children: /* @__PURE__ */ e(F, { children: /* @__PURE__ */ e(L, { style: { height: 368 }, data: c, itemContent: (n, a) => {
            var r;
            return /* @__PURE__ */ i(S, { children: [
              h && c.length === ((r = o.current) == null ? void 0 : r.length) && l[n] && /* @__PURE__ */ e(m, { mt: n === 0 ? "0" : "4", mb: "2", children: /* @__PURE__ */ e(k, { fontWeight: "600", color: "text-secondary", mb: "2", children: l[n] }) }),
              /* @__PURE__ */ e(b, { ...a })
            ] });
          } }) }) })
        ]
      }
    )
  ] });
}
export {
  z as RowList
};
