// @argent/x-ui/dist/src/components/CompoundDialogBottomSheet/components/SlippageInput.js
import { jsxs as m, Fragment as f, jsx as i } from "react/jsx-runtime";
import { Flex as p, IconButton as h, InputGroup as g, Input as x, Box as b, Button as S } from "@chakra-ui/react";
import { useState as y } from "react";
import I from "../../icons/MinusPrimaryIcon.js";
import w from "../../icons/PlusPrimaryIcon.js";
import { Header as C, Content as W, Footer as k } from "./base.js";
function s(r, n) {
  return r < n.min || r > n.max;
}
function d(r, n, o) {
  const e = Number.isNaN(r) ? 0 : r, t = Math.round(n ? e + o.step : e - o.step);
  return s(t, o) ? e : t;
}
function j({ onSave: r, value: n, options: o }) {
  const e = o || {
    multiplier: 100,
    min: 0,
    max: 5e3,
    step: 10
  }, [t, a] = y(n), u = s(t, e);
  return /* @__PURE__ */ m(f, { children: [
    /* @__PURE__ */ i(C, { leftAligned: !0, children: "Edit slippage" }),
    /* @__PURE__ */ i(W, { children: /* @__PURE__ */ i(p, { width: "100%", direction: "column", gap: "6", children: /* @__PURE__ */ m(p, { width: "100%", gap: "3", children: [
      /* @__PURE__ */ i(h, { "aria-label": "Decrease slippage", minWidth: "12", colorScheme: "secondary", isDisabled: t <= e.min, onClick: () => a((l) => d(l, !1, e)), children: /* @__PURE__ */ i(I, {}) }),
      /* @__PURE__ */ m(g, { as: p, alignItems: "center", justifyContent: "center", textAlign: "center", height: "auto", minHeight: "auto", borderRadius: "full", borderWidth: "1px", borderColor: "stroke-default-web", py: "1", px: "3", fontSize: "20px", fontWeight: "500", bg: "surface-elevated", children: [
        /* @__PURE__ */ i(x, { type: "number", placeholder: "0", p: "0", border: "none", height: "auto", minHeight: "auto", textAlign: "center", fontSize: "inherit", fontWeight: "inherit", bg: "inherit", width: "5ch", ml: "1.5ch", step: e.step / e.multiplier, min: e.min, max: e.max, value: t / e.multiplier, onChange: (l) => {
          let c = parseFloat(l.target.value) * e.multiplier;
          c = Math.min(Math.round(c), e.max), a(c);
        }, _focusVisible: {
          outline: "none",
          boxShadow: "none"
        } }),
        /* @__PURE__ */ i(b, { color: "text-primary", fontSize: "inherit", fontWeight: "inherit", children: "%" })
      ] }),
      /* @__PURE__ */ i(h, { "aria-label": "Increase slippage", minWidth: "12", colorScheme: "secondary", isDisabled: t >= e.max, onClick: () => a((l) => d(l, !0, e)), children: /* @__PURE__ */ i(w, {}) })
    ] }) }) }),
    /* @__PURE__ */ i(k, { children: /* @__PURE__ */ i(S, { colorScheme: "primary", onClick: () => r(t), isDisabled: u, children: "Save" }) })
  ] });
}
export {
  j as SlippageInput
};
