// @argent/x-ui/dist/src/components/transactions/TransactionReview/FeeEstimationBox.js
import { jsx as e, jsxs as t } from "react/jsx-runtime";
import { Flex as r } from "@chakra-ui/react";
import "../../Typography.js";
import { Label2Bold as i } from "../../typography.generated.js";
const u = (o) => /* @__PURE__ */ e(r, { borderRadius: "xl", backgroundColor: "white", border: "1px", borderColor: "transparent", pl: 3, pr: 2, py: 2, _dark: {
  backgroundColor: "neutrals.900",
  borderColor: "neutrals.500",
  boxShadow: "menu"
}, ...o }), c = ({ children: o }) => /* @__PURE__ */ t(r, { borderRadius: "xl", border: "1px", flexDir: "column", borderColor: "text-subtle", children: [
  /* @__PURE__ */ e(r, { px: 3, py: 2, children: o }),
  /* @__PURE__ */ e(r, { backgroundColor: "stroke-default-web", justifyContent: "center", alignItems: "center", borderBottomRadius: "xl", p: "1", children: "Includes one-time activation fee" })
] }), x = ({ children: o, userClickedAddFunds: n }) => /* @__PURE__ */ t(r, { borderRadius: "xl", border: "1px", flexDir: "column", borderColor: "surface-danger-vibrant", children: [
  /* @__PURE__ */ e(r, { px: 3, py: 2, children: o }),
  /* @__PURE__ */ e(r, { backgroundColor: "surface-danger-default", justifyContent: "center", alignItems: "center", borderBottomRadius: "xl", p: "1", children: /* @__PURE__ */ e(i, { color: "text-danger", children: n ? "Waiting for funds..." : "Insufficient funds to pay fee" }) })
] });
export {
  u as FeeEstimationBox,
  c as FeeEstimationBoxWithDeploy,
  x as FeeEstimationBoxWithInsufficientFunds
};
